test/integration/framework/master_utils.go

```diff
@@ -155,24 +155,29 @@ func NewMasterConfig() *master.Config {
 		Prefix:     etcdtest.PathPrefix(),
 	}
 
-	negotiatedSerializer := NewSingleContentTypeSerializer(api.Scheme, testapi.Default.Codec(), "application/json")
+	negotiatedSerializer := NewSingleContentTypeSerializer(api.Scheme, testapi.Default.Codec(), runtime.ContentTypeJSON)
 
-	storageFactory := genericapiserver.NewDefaultStorageFactory(config, negotiatedSerializer, genericapiserver.NewDefaultResourceEncodingConfig(), master.DefaultAPIResourceConfigSource())
+	storageFactory := genericapiserver.NewDefaultStorageFactory(config, runtime.ContentTypeJSON, negotiatedSerializer, genericapiserver.NewDefaultResourceEncodingConfig(), master.DefaultAPIResourceConfigSource())
 	storageFactory.SetSerializer(
 		unversioned.GroupResource{Group: api.GroupName, Resource: genericapiserver.AllResources},
-		NewSingleContentTypeSerializer(api.Scheme, testapi.Default.Codec(), "application/json"))
+		"",
```

Comment:
Updated with the cleaner recognizer logic.

On Thu, May 5, 2016 at 11:14 AM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> Actually this is the override type - that bypasses the behavior of the
> serializer itself.  If empty, no override.
>
> On Thu, May 5, 2016 at 3:47 AM, Wojciech Tyczynski <
> notifications@github.com> wrote:
>
>> In test/integration/framework/master_utils.go
>> <https://github.com/kubernetes/kubernetes/pull/24710#discussion_r62157142>
>> :
>>
>> >  	storageFactory.SetSerializer(
>> >  		unversioned.GroupResource{Group: api.GroupName, Resource: genericapiserver.AllResources},
>> > -		NewSingleContentTypeSerializer(api.Scheme, testapi.Default.Codec(), "application/json"))
>> > +		"",
>>
>> Just question: what does empty media type mean?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/24710/files/e416be61b90b5dc461dc8a84a94d6ecd0d76c216#r62157142>
>>
>
>